### PR TITLE
Dtc 20731 defect count by rca

### PR DESCRIPTION
**Pull request by ashutoshy14**
DtS 20731 defect count by rca with UI integration added

### CODE REVIEW

File: UI/src/environments/environment.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@
```
Comment (hirbabar) [inline UI/src/environments/environment.ts:22]: revert this local file changes

File: customapi/src/main/resources/application-dev.properties [hunk 2]
```diff
@@ -17,11 +17,11 @@
 ################################################################################
```
Comment (hirbabar) [inline customapi/src/main/resources/application-dev.properties:18]: revert this local file changes

File: customapi/src/main/resources/application.properties [hunk 3]
```diff
@@ -17,11 +17,11 @@
 ################################################################################
 
 ## MongoDB related properties
-spring.data.mongodb.host=mongodb
+spring.data.mongodb.host=localhost
 spring.data.mongodb.port=27017
```
Comment (hirbabar) [inline customapi/src/main/resources/application.properties:21]: revert this local file changes

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DefectCountByRCAServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
+package com.publicissapient.kpidashboard.apis.jira.scrum.service;
+
+import java.util.*;
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DefectCountByRCAServiceImpl.java:3]: import and file formatting

**Review by hirbabar — APPROVED** [review]

**Review by hirbabar — CHANGES_REQUESTED** [review]
resolved conflicts

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DefectCountByRCAServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+					issueTypes.add(issueType);
+					List<IterationKpiData> data = new ArrayList<>();
+					issues.forEach((key, values) -> {
+						List<IterationKpiModalValue> modalValues = new ArrayList<>();
+						IterationKpiData issueCounts = new IterationKpiData(key, Double.valueOf(values.size()), null, key, "", modalValues);
+						if (values instanceof ArrayList) {
+							for (JiraIssue jiraIssue : values) {
+								overAllIssueCount.set(0, overAllIssueCount.get(0) + 1);
+								populateIterationData(overAllmodalValues, modalValues, jiraIssue);
+							}
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DefectCountByRCAServiceImpl.java:147]: needed to proper dev testing

**Review by hirbabar — CHANGES_REQUESTED** [review]

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DefectCountByRCAServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+		return null;
+	}
+
+	@Override
+	public Map<String, Object> fetchKPIDataFromDb(List<Node> leafNodeList, String startDate, String endDate,
+			KpiRequest kpiRequest) {
+		Map<String, Object> resultListMap = new HashMap<>();
+		Node leafNode = leafNodeList.stream().findFirst().orElse(null);
+		if (null != leafNode) {
+			LOGGER.info("Closure Possible Today -> Requested sprint : {}", leafNode.getName());
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/DefectCountByRCAServiceImpl.java:63]: define meaningful variable name for ex. variable name should be totalIssues

**Review by ashutoshy14 — COMMENTED** [review]
Review changes

**Pull request closed without merging**